python/ray/tests/test_set_task_returns.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import pytest
+
+import ray
+import ray.exceptions
+import ray.experimental.no_return
+import ray.worker
+
+
+@pytest.fixture
+def ray_start():
+    # Start the Ray processes.
+    ray.init(num_cpus=1)
+    yield None
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+def test_set_single_output(ray_start):
+    @ray.remote
+    def f():
+        return_object_ids = ray.worker.global_worker._current_task.returns()
+        ray.worker.global_worker.put_object(return_object_ids[0], 123)
+        return ray.experimental.no_return.NoReturn
+
+    assert ray.get(f.remote()) == 123
+
+
+@pytest.mark.parametrize("set_out0", [True, False])
+@pytest.mark.parametrize("set_out1", [True, False])
+@pytest.mark.parametrize("set_out2", [True, False])
+def test_set_multiple_outputs(ray_start, set_out0, set_out1, set_out2):
```

Comment:
Let's just run all 8 cases inside of a single test. Each call to `ray.init(); ray.shutdown()` takes 1 or 2 seconds, so we don't want to repeat that 8 times unnecessarily.